src/mui/input/NumberInput.js

```diff
@@ -15,21 +15,35 @@ import FieldTitle from '../../util/FieldTitle';
  * The object passed as `options` props is passed to the material-ui <TextField> component
  */
 class NumberInput extends Component {
-    /**
-     * Necessary because of a React bug on <input type="number">
-     * @see https://github.com/facebook/react/issues/1425
-     */
-    handleBlur = () => {
-        this.props.input.onChange(parseFloat(this.props.input.value));
+    handleBlur = (eventOrValue) => {
+        this.props.onBlur(eventOrValue);
+        this.props.input.onBlur(eventOrValue);
+
+        /**
+         * Necessary because of a React bug on <input type="number">
+         * @see https://github.com/facebook/react/issues/1425
+         */
+        this.handleChange(parseFloat(this.props.input.value));
+    }
+
+    handleFocus = (event) => {
```

Comment:
This is voluntarily `event`, not `eventOrValue`, as specified in ReduxForm documentation: http://redux-form.com/6.2.0/docs/api/Field.md/